### PR TITLE
feat(ct): expose `msw` fixture that accepts msw-style request handlers

### PR DESCRIPTION
**Pull request by dgozman**
Implemented on top of `context.route()`.

### CODE REVIEW

File: docs/src/test-components-js.md [hunk 1]
```diff
@@ -721,6 +721,45 @@ test('update', async ({ mount }) => {
 
 </Tabs>
 
+### Handling network requests
+
+Playwright provides [`method: Page.route`] method to route your network requests. See the [network mocking guide](./mock.md) for more details.
+
+### MSW support
+
+If you are using the [MSW library](https://mswjs.io/) to handle network requests during development or testing, try the `msw` fixture that Playwright provides. It supports the same API so that you can reuse your handlers in the test.
```
Comment (pavelfeldman) [inline docs/src/test-components-js.md:730]: Let's say that we are experimenting with it?

**Review by pavelfeldman — APPROVED** [review]

File: packages/playwright-ct-core/src/msw.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  const requestId = String(++lastRequestId);
+  const resolutionContext = { baseUrl };
+  for (const handler of handlers) {
+    const result = await handler.run({ request, requestId, resolutionContext });
+    if (result?.response)
+      return result.response;
+  }
+}
+
+async function globalFetch(...args: Parameters<typeof globalThis.fetch>) {
```
Comment (kettanaito) [inline packages/playwright-ct-core/src/msw.ts:42]: If you add `msw` as a dependency, it exposes the [`getResponse`](https://github.com/mswjs/msw/blob/bed402cdd9b79ef3084b3195cdcce0d83c7e2cfc/src/core/getResponse.ts#L12) function that you can reuse here. Less logic duplication.

File: packages/playwright-ct-core/src/msw.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+ * limitations under the License.
+ */
+
+import type * as playwright from 'playwright/test';
+
+interface RequestHandler {
+  run(args: { request: Request, resolutionContext: { baseUrl?: string }, requestId: string }): Promise<{ response: Response } | undefined>;
+}
+
+export interface MSW {
```
Comment (kettanaito) [inline packages/playwright-ct-core/src/msw.ts:23]: Would be nice to actually `import { RequestHandler } from 'msw'`.

**Comment by kettanaito** [comment]
I have some concerns about this. Please see https://github.com/microsoft/playwright/issues/30981. 

TL;DR I'd rather people used the _same_ MSW experience than a compatible API. MSW works great in any browser, including browsers automated by Playwright. I see no reason to diverge the behavior, losing a significant chunk of MSW's benefits along the way.
